server/index.js

```diff
@@ -1,24 +1,52 @@
-const express = require("express");
-const mongoose = require("mongoose");
-
-const compile = require("./routes/api/compile");
-const zip = require("./routes/api/zip");
-const generate = require("./routes/api/generate");
+const express = require('express');
+const mongoose = require('mongoose');
 
+const compile = require('./routes/api/compile');
+const zip = require('./routes/api/zip');
+const generate = require('./routes/api/generate');
+const upload = require('express-fileupload');
 const app = express();
-
+var fs = require('fs');
 //Database config goes here
 
 //Database connection goes here
-
-app.get("/", (req, res) => {
-  res.send("Hello World");
+app.use(upload());
+app.get('/', (req, res) => {
+	res.send('Hello World');
+});
+app.get('/fileupload', function(req, res) {
+	console.log(__dirname);
+	res.sendFile(__dirname + '/index.html');
+});
+app.post('/upload', function(req, res) {
+	console.log(req.files);
+	var file = req.files.upfile;
+	if (file) {
+		var filename = file.name;
+		var timestamp = new Date().getTime();
+		var uploadpath = __dirname + '/uploads/' + filename + timestamp;
```

Comment:
This changes the file format.
```suggestion
              var uploadpath = __dirname + '/uploads/' + timestamp + "-" + filename;
```